File: planning-idsdp-pps-frontend/src/app/main/modules/dpp-tapp-management/services/tapp-financing-and-expectation.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {CrudRequestService} from '../../../core/services/crud-request.service';
import {environment} from '../../../../../environments/environment';
import {TappFinancingAndExpectationModel} from '../models/tappFinancingAndExpectation.model';
import {map} from "rxjs/operators";

@Injectable({
  providedIn: 'root'
})
export class TappFinancingAndExpectationService extends CrudRequestService<TappFinancingAndExpectationModel>{

    constructor(private http: HttpClient) {
        super(http, environment.ibcs.ppsDppBackendPoint + 'financing-and-expectation/');
    }

    getFinancingExpectation(pcUuid){
        const url: string = environment.ibcs.ppsDppBackendPoint + 'financing-and-expectation/get-data' + '/' + pcUuid;
        return this.http.get(url).pipe(map((res: any) => {
            return res;
        }));
    }

    updateFinancingExpectation(model, pcUuid){
        const url: string = environment.ibcs.ppsDppBackendPoint + 'financing-and-expectation/update-data' + '/' + pcUuid;
        return this.http.put(url, model).pipe(map((res: any) => {
            return res;
        }));
    }


    createFinancingExpectation(model){
        const url: string = environment.ibcs.ppsDppBackendPoint + 'financing-and-expectation/create-financing-expectation';
        return this.http.post(url, model).pipe(map((res: any) => {
            return res;
        }));
    }
}

File: planning-idsdp-pps-dpp-backend/src/main/java/com/ibcs/idsdp/dpptapp/services/TappFinancingAndExpectationService.java
package com.ibcs.idsdp.dpptapp.services;

import com.ibcs.idsdp.common.web.dto.response.ResponseWithResults;
import com.ibcs.idsdp.dpptapp.model.domain.TappFinancingAndExpectation;
import com.ibcs.idsdp.dpptapp.web.dto.TappFinancingAndExpectationDTO;
import com.ibcs.idsdp.dpptapp.web.dto.TappModeFinancingDTO;

public interface TappFinancingAndExpectationService {
    ResponseWithResults getFinancingExpectation(String pcUuid);
    TappFinancingAndExpectation saveFinancingExpectation(TappFinancingAndExpectationDTO dto);
    TappFinancingAndExpectation updateFinancingExpectation(TappFinancingAndExpectationDTO dto, String pcUuid);
}
